host/class/uvc/usb_host_uvc_2/uvc_host.c

```diff
@@ -0,0 +1,816 @@
+}
+
+esp_err_t uvc_host_stream_stop(uvc_host_stream_hdl_t stream_hdl)
+{
+    UVC_CHECK(stream_hdl, ESP_ERR_INVALID_ARG);
+    uvc_stream_t *uvc_stream = (uvc_stream_t *)stream_hdl;
+
+    ESP_RETURN_ON_ERROR(uvc_host_stream_pause(stream_hdl), TAG, "Could not pause the stream");
+
+    if (uvc_stream->constant.bAlternateSetting != 0) { // if (is_isoc_stream)
+        // ISOC streams are stopped by setting alternate interface 0
+        return uvc_set_interface(stream_hdl, false);
+    } else {
+        // BULK streams are stopped by halting the endpoint
+        return uvc_clear_endpoint_feature(stream_hdl);
+    }
+}
+
+esp_err_t uvc_host_stream_pause(uvc_host_stream_hdl_t stream_hdl)
+{
+    UVC_CHECK(stream_hdl, ESP_ERR_INVALID_ARG);
+    uvc_stream_t *uvc_stream = (uvc_stream_t *)stream_hdl;
+
+    // We do not cancel the ongoing transfers here, it is not supported by USB Host Library
+    // By setting uvc_stream->dynamic.streaming = false; no frame callbacks will be called and the transfer can gracefully finish
+    UVC_ENTER_CRITICAL();
+    UVC_CHECK_FROM_CRIT(uvc_stream->dynamic.streaming, ESP_OK); // Return immediately if already paused
+    uvc_stream->dynamic.streaming = false;
+    uvc_host_frame_t *current_frame = uvc_stream->dynamic.current_frame;
+    uvc_stream->dynamic.current_frame = NULL;
+    UVC_EXIT_CRITICAL();
+
+    if (current_frame) {
+        uvc_host_frame_return(uvc_stream, current_frame);
+    }
+
+    //@todo this is not a clean solution
+    vTaskDelay(pdMS_TO_TICKS(50)); // Wait for all transfers to finish
+    return ESP_OK;
+}
+
+esp_err_t uvc_host_stream_unpause(uvc_host_stream_hdl_t stream_hdl)
+{
+    UVC_CHECK(stream_hdl, ESP_ERR_INVALID_ARG);
+    uvc_stream_t *uvc_stream = (uvc_stream_t *)stream_hdl;
+    esp_err_t ret = ESP_OK;
+
+    UVC_ENTER_CRITICAL();
+    UVC_CHECK_FROM_CRIT(!uvc_stream->dynamic.streaming, ESP_ERR_INVALID_STATE);
+    uvc_stream->dynamic.streaming = true;
+    // Start of Frame is detected when received FrameID != current_frame_id
+    // We set current_frame_id to illegal value (FrameID can be 0 or 1) so we catch SoF of the very first frame
+    uvc_stream->single_thread.current_frame_id = 2;
+    uvc_stream->single_thread.next_bulk_packet = UVC_STREAM_BULK_PACKET_SOF;
+    UVC_EXIT_CRITICAL();
+
+    for (int i = 0; i < uvc_stream->constant.num_of_xfers; i++) {
+        ESP_GOTO_ON_ERROR(
+            usb_host_transfer_submit(uvc_stream->constant.xfers[i]),
+            stop_stream, TAG, "Could not submit transfer %d", i);
+    }
+    return ret;
+
+stop_stream:
+    uvc_host_stream_pause(stream_hdl);
+    return ret;
+}
+
+static void ctrl_xfer_cb(usb_transfer_t *transfer)
+{
+    ESP_LOGD(TAG, "ctrl xfer cb");
+    assert(transfer->context);
+    xSemaphoreGive((SemaphoreHandle_t)transfer->context);
+}
+
+esp_err_t uvc_host_usb_ctrl(uvc_host_stream_hdl_t stream_hdl, uint8_t bmRequestType, uint8_t bRequest, uint16_t wValue, uint16_t wIndex, uint16_t wLength, uint8_t *data)
```

Comment:
Do we plan to support something else rather then usb? 
If no, maybe `usb` in not necessary to be in the name.

```suggestion
esp_err_t uvc_host_ctrl(uvc_host_stream_hdl_t stream_hdl, uint8_t bmRequestType, uint8_t bRequest, uint16_t wValue, uint16_t wIndex, uint16_t wLength, uint8_t *data)
```

Reply:
Then "ctrl_request" probably would be a bit better option....
```suggestion
esp_err_t uvc_host_ctrl_request(uvc_host_stream_hdl_t stream_hdl, uint8_t bmRequestType, uint8_t bRequest, uint16_t wValue, uint16_t wIndex, uint16_t wLength, uint8_t *data)
```